source/databricks/package/balance_fixing_total_production.py

```diff
@@ -184,9 +188,115 @@ def _get_grid_areas_df(cached_integration_events_df, batch_grid_areas_df) -> Dat
     return grid_area_df
 
 
+def _get_master_basis_data_df(
+    metering_points_periods_df: DataFrame,
+    market_roles_periods_df: DataFrame,
+    grid_area_df: DataFrame,
+    period_start_datetime,
+    period_end_datetime,
+) -> DataFrame:
+    metering_points_in_grid_area = metering_points_periods_df.join(
+        grid_area_df,
+        metering_points_periods_df["GridArea"] == grid_area_df["GridAreaCode"],
+        "inner",
+    )
+
+    metering_point_periods_df = (
+        metering_points_in_grid_area.where(col("FromDate") <= period_end_datetime)
+        .where(col("ToDate") >= period_start_datetime)
+        .where(
+            (col("ConnectionState") == "E22") | (col("ConnectionState") == "E23")
+        )  # Connected or Disconnected
+        .where(col("MeteringPointType") == "E18")  # Production
+    )
+
+    market_roles_periods_df = market_roles_periods_df.where(
+        col("FromDate") <= period_end_datetime
+    ).where(col("ToDate") >= period_start_datetime)
```

Comment:
Shouldn't it be like this?
```suggestion
        col("FromDate") < period_end_datetime
    ).where(col("ToDate") > period_start_datetime)
```

Reply:
has been changed